wire-ios-sync-engine/Tests/Source/Repositories/ConnectionsRepositoryTests.swift

```diff
@@ -0,0 +1,154 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+@testable import WireAPI
+import WireAPISupport
+import WireDataModelSupport
+@testable import WireSyncEngine
+import WireSyncEngineSupport
+import XCTest
+
+final class ConnectionsRepositoryTests: XCTestCase {
+
+    var sut: ConnectionsRepository!
+    var connectionsAPI: MockConnectionsAPI!
+
+    var stack: CoreDataStack!
+    let coreDataStackHelper = CoreDataStackHelper()
+    let modelHelper = ModelHelper()
+
+    var context: NSManagedObjectContext {
+        stack.syncContext
+    }
+
+    override func setUp() async throws {
+        try await super.setUp()
+        BackendInfo.storage = .temporary()
+
+        stack = try await coreDataStackHelper.createStack()
+        connectionsAPI = MockConnectionsAPI()
+        sut = ConnectionsRepository(connectionsAPI: connectionsAPI,
+                                    context: context)
+    }
+
+    override func tearDown() async throws {
+        stack = nil
+        connectionsAPI = nil
+        sut = nil
+        try coreDataStackHelper.cleanupDirectory()
+        BackendInfo.storage = .standard
+        try await super.tearDown()
+    }
+
+    // MARK: - Tests
+
+    func testPullConnections_GivenOneConnectionFails_OtherConnectionsAreStored() async throws {
+        // Mock
+        let connection = Scaffolding.connection
+        let brokenConnection = WireAPI.Connection(senderId: nil,
+                                                  receiverId: nil,
+                                                  receiverQualifiedId: nil,
+                                                  conversationId: nil,
+                                                  qualifiedConversationId: nil,
+                                                  lastUpdate: Date(),
+                                                  status: .pending)
+
+        connectionsAPI.getConnections_MockValue = .init(fetchPage: { _ in
+
+            return WireAPI.PayloadPager.Page(element: [brokenConnection, connection], hasMore: false, nextStart: "first")
```

Comment:
There are some more of these kind.. usually with more than 2 parameters line breaks help to read:
```suggestion
            return WireAPI.PayloadPager.Page(
                element: [
                    brokenConnection,
                    connection
                ],
                hasMore: false,
                nextStart: "first"
            )
```

Reply:
depends on your screen ;)